backend/de.metas.business/src/main/java/org/eevolution/api/impl/ProductBOMCycleDetection.java

```diff
@@ -12,44 +16,48 @@
 import javax.swing.tree.DefaultMutableTreeNode;
 import java.util.LinkedHashSet;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 
+import static org.compiere.util.Env.getCtx;
+
 /**
- * Calculates product's low level code (LLC) and also checks for BOM cycles (it is throwing {@link BOMCycleException} in that case).
- * 
+ * Checks for BOM cycles (it is throwing {@link BOMCycleException} in that case).
+ *
  * @author metas-dev <dev@metasfresh.com>
  */
-/* package */class ProductLowLevelCalculator
+class ProductBOMCycleDetection
 {
-	public static ProductLowLevelCalculator newInstance()
+	public static ProductBOMCycleDetection newInstance()
 	{
-		return new ProductLowLevelCalculator();
+		return new ProductBOMCycleDetection();
 	}
-
+	private static final String ERR_PRODUCT_BOM_CYCLE = "Product_BOM_Cycle_Error";
 	private final Set<ProductId> seenProductIds = new LinkedHashSet<>();
 
-	private ProductLowLevelCalculator()
+	private ProductBOMCycleDetection()
 	{
 	}
 
-	/**
-	 * Calculate the low level of given product
-	 */
-	public int getLowLevel(final ProductId productId)
+	public void checkCycles(final ProductId productId)
 	{
-		clearSeenProducts();
-		markProductAsSeen(productId);
-
-		final DefaultMutableTreeNode ibom = createParentProductNode(productId); // start traversing tree
-		return ibom.getDepth();
+		try
+		{
+			assertNoCycles(productId);
+		}
+		catch (final BOMCycleException e)
+		{
+			final I_M_Product product = Services.get(IProductBL.class).getById(productId);
+			throw new AdempiereException(Services.get(IMsgBL.class).getMsg(getCtx(), ERR_PRODUCT_BOM_CYCLE, new Object[] { product.getValue() }));
+		}
 	}
 
 	/**
-	 * get an implotion the product
-	 * 
+	 * get an implosion of the product
+	 *
 	 * @return DefaultMutableTreeNode Tree with all parent product
 	 */
-	private DefaultMutableTreeNode createParentProductNode(final ProductId productId)
+	private DefaultMutableTreeNode assertNoCycles(final ProductId productId)
```

Comment:
completelly remove the javadoc. it's stale and pointless

Reply:
done